streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -860,32 +860,37 @@ public KafkaStreams(final Topology topology,
     public KafkaStreams(final Topology topology,
                         final StreamsConfig applicationConfigs,
                         final Time time) {
-        this(new TopologyMetadata(topology.internalTopologyBuilder, applicationConfigs), applicationConfigs, applicationConfigs.getKafkaClientSupplier(), time);
+        this(topology, new InternalStreamsConfig(applicationConfigs), null, time);
```

Comment:
Some side cleanup on constructor methods -- no public change -- just to make the flow simpler